packages/remix-react/components.tsx

```diff
@@ -210,6 +220,9 @@ let NavLink = React.forwardRef<HTMLAnchorElement, RemixNavLinkProps>(
           {...prefetchHandlers}
           ref={mergeRefs(forwardedRef, ref)}
           to={to}
+          data-discover={
+            !isAbsolute && discover === "render" ? "true" : undefined
+          }
```

Comment:
`data-discover` opts into the manifest patching.  It's enabled by default (`discover="render"`), users can opt out via `<Link discover="click">`/`<NavLink discover="click">`

Reply:
The use of a `data-` attribute here instead of just react state is for a few reasons:

1. Makes it easier to detect via `MutationObserver` and not a mess of React context/effects where Links "register" themselves with some parent context
2. It scales very well into an RSC world where you may just want to render `<a href="">` and not _have_ to use a client component just to participate in fog of war
3. If we ever remove `Link` in favor of event delegation, it works there too